central/role/datastore/datastore_test.go

```diff
@@ -386,6 +393,29 @@ func (s *roleDataStoreTestSuite) TestPermissionSetWriteOperations() {
 
 	err = s.dataStore.AddPermissionSet(s.hasWriteCtx, goodPermissionSet)
 	s.NoError(err, "adding a permission set with ID and name that used to exist is not an error")
+
+	err = s.dataStore.AddPermissionSet(s.hasWriteCtx, declarativePermissionSet)
+	s.NoError(err, "adding a permission set declaratively is not an error")
```

Comment:
General point:
Just from a consistency point of view, should we also limit _adding_ declarative resources via the context?

Realistically, only the declarative config manager should be the one allowed to add declarative resources - otherwise users may add declarative resources but essentially can never delete them / do anything with them from the API POV.

On the other hand, the declarative resources should be reconciled and will be deleted in an eventual consistent manner, but from the UX point of view these things might be confusing.

Let's add a follow-up to the current story, since it doesn't include the whole "constrain API interactions with declarative resources to a particular context / client" and constrain the ability to add declarative resources within there. WDYT?

Reply:
Sounds good. Created a follow up ticket: https://issues.redhat.com/browse/ROX-14999